### ISSUE TITLE
Popups configured with invalid zero size

### ISSUE THREAD

**Issue by wmww** [MirServer/mir#836]
```
43.8434 → xdg_popup@46.23 configure(x=-4, y=50, width=0, height=0)
43.8445 → xdg_surface@34.28 configure(serial=223)
43.8772 xdg_surface@34.28 ack_configure(serial=223) ↲
43.9568 xdg_surface@34.28 set_window_geometry(x=4, y=4, width=0, height=0) ↲
44.0178 → wl_display@1.0 error(object_id=wl_display@1.0, code=2, message='no memory')
[2019-05-21 14:21:44.637417] < - ERROR - > frontend:Wayland: Exception processing Surface::commit() request: /home/wmww/code/mir/src/miral/basic_window_manager.cpp(2021): Throw in function void miral::BasicWindowManager::validate_modification_request(const miral::WindowSpecification&, const miral::WindowInfo&) const
Dynamic exception type: boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<std::runtime_error> >
std::exception::what: width must be positive
```
As you can see in this wayland-debug snippet from Mir interacting with Gedit, a popup is configured with zero size, gedit then sets its window geometry size to zero in response, which causes Mir to choke and disconnect it. Several things need to be fixed:
* Mir shouldn't configure popups with zero size (it doesn't seem to be explicitly specified as valid or invalid, but since GTK responds by setting window geometry to zero size (a protocol violation), lets assume we shouldn't do it.
* Mir should detect the zero size window geometry and take deliberate action, instead of feeding it into the window management system and get an exception
* Mir should send the appropriate error to the client, instead of "no memory"

**Comment by wmww** [comment]
Additional reading of the protocol revels that `xdg_toplevel.configure()` sizes (in both v6 and stable) *are* allowed to be zero. This is presumably why we were doing it in the first place, then the same logic got ported to popups without thinking about it too much. I also found a hack to get around this crash in gtk-layer-shell (which is obviously the wrong place to fix it).

**Comment by AlanGriffiths** [comment]
As Mir now responds by disconnecting the client this should probably get at least a workaround for 1.2.

**Comment by AlanGriffiths** [comment]
What I lack is a reliable "reproducer". @wmww how do you get this to occur?

**Comment by wmww** [comment]
I can reproduce it reliably as follows:
* Open gedit
* Hover over the `Open` button until the tooltip appears
* The tooltip appears for one frame, then gedit closes

**Comment by wmww** [comment]
Fixed by #838, not sure why this didn't auto-close.
